lib/autoload/globals/Iter.ngs

```diff
@@ -373,3 +373,48 @@ F skip(i:Iter, count:Int){
 TEST [3, 4, 5, 34, 5].Iter().skip(3).Arr() == [34, 5] 
 TEST [3, 40, 15, 34, 5].Iter().skip(2).Arr() == [15, 34, 5] 
 TEST {[3, 40, 15, 34, 5].Iter().skip(10)}.assert(SkipError)
+
+doc Type that allows us to iterate over a directory.
+type DirIter(Iter)
+
+# ---------- DirIter ----------
+doc Calls DirIter constructor.
+doc %RET - DirIter which allows us to iterate over a directory
+F Iter(d:Dir) DirIter(d)
+
+doc Initialize the DirIter with specific Dir
+F init(i:DirIter, d:Dir){
+	i.val = d
+	i.paths = dir(d)
+	i.idx = 0
+}
+
+doc Check whether there are more paths to iterate over.
+doc %RET - Bool. true: next() will return next element. false: next() will throw NoNext
+F Bool(i:DirIter) i.idx < i.paths.len()
+
+doc Get next path of a directory iterated over.
+doc %RET - Path
+F next(d:DirIter) {
+	throw_if_no_next(d)
+	ret = d.paths[d.idx]
+	d.idx += 1
+	ret
+}
+
+doc Method that allows us to iterate the specified Dir()
+doc %RET - DirIter
+F Iter(d:Dir){
+	i = DirIter(d)
+	i
+}
+
+doc Calls cb with each path from the directory
+doc %RET - i
+F each(d:Dir, cb:Fun){
+	i = DirIter(d)
+	for e in i {
+		cb(e)
+	}
+	i
+}
```

Comment:
Should reuse existing `each(i:Iter, cb:Fun)`.

So something like `F each(d:Dir, cb:Fun) d.Iter().each(cb)` (untested)

Reply:
is this correct?